src/data-loader.ts

```
@@ -935,17 +1013,27 @@ export async function loadMonthlyUsageData(
 ): Promise<MonthlyUsage[]> {
 	const dailyData = await loadDailyUsageData(options);
 
-	// Group daily data by month using Object.groupBy
-	const groupedByMonth = groupBy(dailyData, data =>
-		data.date.substring(0, 7));
+	// Group daily data by month, optionally including project
+	// Automatically enable project grouping when project filter is specified
+	const needsProjectGrouping = options?.groupByProject === true || options?.project != null;
+	const groupingKey = needsProjectGrouping
+		? (data: typeof dailyData[0]) => `${data.date.substring(0, 7)}|${data.project ?? 'unknown'}`
```

ESLint

[error] Unsafe call of a(n) `error` type typed value.

(ts/no-unsafe-call)